security/pkg/nodeagent/sds/server.go

```diff
@@ -44,6 +50,13 @@ type Options struct {
 
 	// The CA provider name.
 	CAProviderName string
+
+	// TrustDomain corresponds to the trust root of a system.
+	// https://github.com/spiffe/spiffe/blob/master/standards/SPIFFE-ID.md#21-trust-domain
+	TrustDomain string
+
+	// PluginNames is plugins' name for certain authN provider.
+	PluginNames []string
```

Comment:
Why this is a list? Do you want to make it support a series of token exchanging, based on the implementation below? I think the first succeeded one is enough?

Reply:
for now yes we only have one, but each authN provider may have one or more plugins(like today's pilot has authn/authz/mixer/etc), put a list here for extensibility; some plugin may do other things rather than token exchange, or a series of token exchanges.

Reply:
However the interface defining `Plugin` requires `ExchangeToken` method instead of something more general like `Do`. I think it's a bit of premature optimization. Although the actual implementation is not complicated. 

Leave it up to you.